src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.AgentPool.ps1

```diff
@@ -445,6 +509,50 @@ class AgentPool: ADOSVTBase
         return $controlResult
     }
 
+    hidden [ControlResult] CheckCredInEnvironmentVariablesAutomatedFix([ControlResult] $controlResult)
+    {
+        try 
+        {     
+            $RawDataObjForControlFix = @();
+            $RawDataObjForControlFix = ([ControlHelper]::ControlFixBackup | where-object {$_.ResourceId -eq $this.ResourceId}).DataObject
+            $FixObjTable = @{}
+            $UndoFixObjTable = @{}
+
+            $RawDataObjForControlFix.UndoFixObj.PSObject.properties | ForEach-Object {  
+                $secureVariableValue = $_.Value | ConvertTo-SecureString
+                $UndoFixObjTable[$_.Name] = [Helpers]::ConvertToPlainText($secureVariableValue);
+            }
+            $RawDataObjForControlFix.FixObj.PSObject.properties | ForEach-Object { $FixObjTable[$_.Name] = $_.Value}
+
+            $display = ($UndoFixObjTable.Keys |  FT -AutoSize | Out-String -Width 512);
+            #The api requires both secret and non secret capabilities in order to update on the agent
+            $UndoFixObjTable += $FixObjTable 
+
+            $RawDataObjForControlFix | ForEach-Object {
+                $CurrentAgent= $_
+                if (-not $this.UndoFix)
+                {                 
+                    $body =  $FixObjTable | ConvertTo-Json   
+                    $controlResult.AddMessage([VerificationResult]::Fixed,  "Following User-defined capabilities for agent pool has been Removed.");      
+                }
+                else 
+                {
+                    $body = $UndoFixObjTable  | ConvertTo-Json
+                    $controlResult.AddMessage([VerificationResult]::Fixed,  "Following User-defined capabilities for agent pool has been Added.");
```

Comment:
user-defined..... have been added: (uniform capitalization and colon in the end)